src/DGtal/geometry/surfaces/estimation/estimationFunctors/SphereFittingEstimator.h

```diff
@@ -0,0 +1,281 @@
+     * \mathbb{R}^3@f$.
+     * @tparam TNormalVectorEstimatorCache the type of normal vector
+     * czche to consider (see EstimatorCache class).
+     */
+    template <typename TSurfel,
+              typename TEmbedder,
+              typename TNormalVectorEstimatorCache>
+    class SphereFittingEstimator
+    {
+    public:
+
+      
+      class PatatePoint
+      {
+      public:
+	enum {Dim = 3};
+	typedef double Scalar;
+	typedef Eigen::Matrix<Scalar, Dim, 1>   VectorType;
+	typedef Eigen::Matrix<Scalar, Dim, Dim> MatrixType;
+        
+	MULTIARCH inline PatatePoint(const VectorType& _pos    = VectorType::Zero(), 
+                                     const VectorType& _normal = VectorType::Zero())
+          : m_pos(_pos), m_normal(_normal) {}
+        
+	MULTIARCH inline const VectorType& pos()    const { return m_pos; }  
+	MULTIARCH inline const VectorType& normal() const { return m_normal; }
+
+	MULTIARCH inline VectorType& pos()    { return m_pos; }  
+	MULTIARCH inline VectorType& normal() { return m_normal; }
+
+      private:
+	VectorType m_pos, m_normal;
+      };
+
+     
+      typedef TSurfel Surfel;
+      typedef TEmbedder SCellEmbedder;
+      typedef typename SCellEmbedder::RealPoint RealPoint;
+
+      typedef TNormalVectorEstimatorCache NormalVectorEstimatorCache;
+
+      typedef typename PatatePoint::Scalar Scalar;
+      typedef typename PatatePoint::VectorType VectorType;
+      
+      typedef Grenaille::DistWeightFunc<PatatePoint,Grenaille::SmoothWeightKernel<Scalar> > WeightFunc; 
+      typedef Grenaille::Basket<PatatePoint,WeightFunc,Grenaille::OrientedSphereFit, Grenaille::GLSParam> Fit;
+
+      ///Quantity type: a 3-sphere (model of CQuantity)
+      struct Quantity
+      {
+        RealPoint center;
+        double radius;
+        double tau;
+        double kappa;
+        RealPoint eta;
+        
+        Quantity(){}
+        Quantity(RealPoint p, double rad, double _tau,
+                 double _kappa, RealPoint _eta): center(p), radius(rad),
+                                                 tau(_tau), kappa(_kappa),
+                                                 eta(_eta) {}
+        ~Quantity(){}
+        bool operator==(Quantity aq) {return (center==aq.center) && (radius==aq.radius);}
+        bool operator<(Quantity aq) {return (center<aq.center) && (radius<aq.radius);}        
+        bool operator!=(Quantity aq) {return !(*this == aq);}
+      };
+
+
+      /**
+       * Constructor.
+       *
+       * @param anEmbedder embedder to map surfel to R^n.
+       * @param h gridstep.
+       */
+      SphereFittingEstimator(ConstAlias<SCellEmbedder> anEmbedder,
+                             const double h,
+                             ConstAlias<NormalVectorEstimatorCache> anEstimator):
+        myEmbedder(&anEmbedder), myH(h), myNormalEsitmatorCache(&anEstimator)
+      {
+        //From Mellado's example
+        myFit = new Fit();
+        myFit->setWeightFunc(WeightFunc(100.0));
```

Comment:
I would say no: this weight function comes from Mellado and I'd prefer to keep this setting from Mellado's documentation. I would agree if you could use DGtal weight functions but it is not the case.